### PR TITLE
chore: use dependabot groups

### PR DESCRIPTION
**Pull request by matifali**
See https://github.blog/changelog/2023-06-30-grouped-version-updates-for-dependabot-public-beta/ for details. This will reduce the number of dependent PRs

For go and node dependencies, I leave the exact grouping to engineers who better know how they should be grouped.

### CODE REVIEW

**Review by coadler — CHANGES_REQUESTED** [review]
I don't really think this is how we should use groups. We want separate commits for each dependency so we can debug them individually, and properly bisect them if they cause a regression.

One thing that I can think of that would benefit from this is our otel deps, which all need to be updated in tandem.
```
	go.nhat.io/otelsql v0.9.0
	go.opentelemetry.io/otel v1.14.0
	go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.14.0
	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.14.0
	go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp v1.14.0
	go.opentelemetry.io/otel/sdk v1.14.0
	go.opentelemetry.io/otel/trace v1.14.0
```

**Comment by ammario** [comment]
If we came across a large dependabot upgrade commit we could still continue the log2(n) bisect, albeit with some manual effort.

**Comment by matifali** [comment]
I removed the super groups for both node and go and added a group for `otel` only
@coadler

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -110,5 +129,7 @@ updates:
     commit-message:
       prefix: "chore"
     labels: []
-    ignore:
-      - dependency-name: "coder/coder"
```
Comment (coadler) [inline .github/dependabot.yaml:114]: Why was this removed?

**Review by coadler — COMMENTED** [review]
Looks like you removed it for GHA and not node

File: .github/dependabot.yaml (hunk 1 above)
Comment (matifali) [reply .github/dependabot.yaml:114]: Because IMO for dogfood we should always have the latest terraform provider. For example templates, we are ignoring and prefer manual updates.

Comment (coadler) [reply .github/dependabot.yaml:114]: Normally changes to our Terraform provider come with added features, changes, or deprecations that need to be added to the example templates as well. I think it makes sense to do these changes at the same time.

Comment (matifali) [reply .github/dependabot.yaml:114]: Sure. I agree. I will revert this.

**Review by coadler — APPROVED** [review]
